### PR TITLE
BlockObverver.OnNextCore must not try to extract deposits from the latest block

### LINKED ISSUE #195: BlockObverver.OnNextCore must not try to extract deposits from the latest block
In this code:
```
        protected override void OnNextCore(ChainedHeaderBlock chainedHeaderBlock)
        {
            this.walletSyncManager.ProcessBlock(chainedHeaderBlock.Block);

            this.blockTipSender.SendBlockTipAsync(
                new BlockTipModel(
                    chainedHeaderBlock.ChainedHeader.HashBlock,
                    chainedHeaderBlock.ChainedHeader.Height,
                    (int)this.depositExtractor.MinimumDepositConfirmations));

            var withdrawals = this.withdrawalExtractor.ExtractWithdrawalsFromBlock(
                chainedHeaderBlock.Block,
                chainedHeaderBlock.ChainedHeader.Height);

            this.withdrawalReceiver.ReceiveWithdrawals(withdrawals);

            IMaturedBlockDeposits maturedBlockDeposits =
                this.depositExtractor.ExtractMaturedBlockDeposits(chainedHeaderBlock.ChainedHeader);

            if (maturedBlockDeposits == null) return;

            this.maturedBlockSender.SendMaturedBlockDepositsAsync(maturedBlockDeposits).ConfigureAwait(false);
        }
    }
```

The method call `ExtractMaturedBlockDeposits` should subtract the deposit confirmations from the block height being parsed for mature block deposits.

### PR DESCRIPTION
**Pull request by quantumagi**
Closes #195

### CODE REVIEW

File: src/Stratis.FederatedPeg.Features.FederationGateway/Interfaces/IDepositExtractor.cs [hunk 1]
```diff
@@ -11,12 +11,37 @@ namespace Stratis.FederatedPeg.Features.FederationGateway.Interfaces
     /// </summary>
     public interface IDepositExtractor
```
Comment (monsieurleberre) [inline src/Stratis.FederatedPeg.Features.FederationGateway/Interfaces/IDepositExtractor.cs]: Thanks for that documentation !

File: src/Stratis.FederatedPeg.Features.FederationGateway/SourceChain/MaturedBlocksProvider.cs [hunk 2]
```diff
@@ -71,7 +71,7 @@ public async Task<List<IMaturedBlockDeposits>> GetMaturedDepositsAsync(int block
```
Comment (monsieurleberre) [inline src/Stratis.FederatedPeg.Features.FederationGateway/SourceChain/MaturedBlocksProvider.cs]: in the code above, is there now a way to reuse the IDepositExtractor's GetNewlyMaturedBlock (just to try avoid duplicating the logic that goes back a few blocks) ?

File: src/Stratis.FederatedPeg.Features.FederationGateway/Interfaces/IDepositExtractor.cs [hunk 3]
```diff
@@ -11,12 +11,37 @@ namespace Stratis.FederatedPeg.Features.FederationGateway.Interfaces
...
+        /// <summary>
+        /// Extracts a deposit from a transaction (if any).
+        /// </summary>
+        /// <param name="transaction">The transaction to extract deposits from.</param>
+        /// <param name="blockHeight">The block height of the block containing the transaction.</param>
+        /// <param name="blockHash">The block hash of the block containing the transaction.</param>
+        /// <returns>The extracted deposit (if any), otherwise <c>null</c>.</returns>
         IDeposit ExtractDepositFromTransaction(Transaction transaction, int blockHeight, uint256 blockHash);
 
+        /// <summary>
```
Comment (monsieurleberre) [inline src/Stratis.FederatedPeg.Features.FederationGateway/Interfaces/IDepositExtractor.cs]: so if I understand, we need both
1) ExtractMaturedBlockDeposits => from a Source chain's point of view (A chain)
2) ExtractBlockDeposits => from a Target node's point of view (B chain)
correct ?

Comment (quantumagi) [reply src/Stratis.FederatedPeg.Features.FederationGateway/Interfaces/IDepositExtractor.cs]: The ExtractBlockDeposits makes a lot more sense from Node B's perspective, or when extracting more than  one block at a time.

File: src/Stratis.FederatedPeg.Features.FederationGateway/SourceChain/MaturedBlocksProvider.cs (hunk 2 above)
Comment (quantumagi) [reply src/Stratis.FederatedPeg.Features.FederationGateway/SourceChain/MaturedBlocksProvider.cs]: We're not really duplicating anything as one of those methods is calling  the other

**Pull request merged**